From da0abb72003fe17bea54a46185d14ed209f6c6c3 Mon Sep 17 00:00:00 2001
From: Soren Brinkmann <soren.brinkmann@xilinx.com>
Date: Thu, 27 Feb 2014 17:54:24 -0800
Subject: [PATCH 112/182] USB: phy-zynq: Remove clock notifier

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

The clock notifier was just an emty placeholder. Remove this dead code.

Signed-off-by: Soren Brinkmann <soren.brinkmann@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 163fb76a5c12cfc432ab19a5491345d1a1077b43)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/usb/phy/phy-zynq-usb.c |   26 --------------------------
 1 files changed, 0 insertions(+), 26 deletions(-)

diff --git a/drivers/usb/phy/phy-zynq-usb.c b/drivers/usb/phy/phy-zynq-usb.c
index f043066..12aff85 100644
--- a/drivers/usb/phy/phy-zynq-usb.c
+++ b/drivers/usb/phy/phy-zynq-usb.c
@@ -1886,25 +1886,6 @@ do_hnp(struct device *dev, struct device_attribute *attr,
 }
 static DEVICE_ATTR(do_hnp, S_IWUSR, NULL, do_hnp);
 
-static int zynq_otg_clk_notifier_cb(struct notifier_block *nb,
-		unsigned long event, void *data)
-{
-
-	switch (event) {
-	case PRE_RATE_CHANGE:
-		/* if a rate change is announced we need to check whether we can
-		 * maintain the current frequency by changing the clock
-		 * dividers.
-		 */
-		/* fall through */
-	case POST_RATE_CHANGE:
-		return NOTIFY_OK;
-	case ABORT_RATE_CHANGE:
-	default:
-		return NOTIFY_DONE;
-	}
-}
-
 static struct attribute *inputs_attrs[] = {
 	&dev_attr_a_bus_req.attr,
 	&dev_attr_a_bus_drop.attr,
@@ -1936,7 +1917,6 @@ static int zynq_otg_remove(struct platform_device *pdev)
 	sysfs_remove_group(&pdev->dev.kobj, &debug_dev_attr_group);
 	device_remove_file(&pdev->dev, &dev_attr_hsm);
 	device_remove_file(&pdev->dev, &dev_attr_registers);
-	clk_notifier_unregister(xotg->clk, &xotg->clk_rate_change_nb);
 	clk_disable_unprepare(xotg->clk);
 
 	return 0;
@@ -1992,11 +1972,6 @@ static int zynq_otg_probe(struct platform_device *pdev)
 		goto err;
 	}
 
-	xotg->clk_rate_change_nb.notifier_call = zynq_otg_clk_notifier_cb;
-	xotg->clk_rate_change_nb.next = NULL;
-	if (clk_notifier_register(xotg->clk, &xotg->clk_rate_change_nb))
-		dev_warn(&pdev->dev, "Unable to register clock notifier.\n");
-
 	/* OTG common part */
 	xotg->dev = &pdev->dev;
 	xotg->otg.dev = xotg->dev;
@@ -2073,7 +2048,6 @@ static int zynq_otg_probe(struct platform_device *pdev)
 	return 0;
 
 err_out_clk_disable:
-	clk_notifier_unregister(xotg->clk, &xotg->clk_rate_change_nb);
 	clk_disable_unprepare(xotg->clk);
 err:
 	zynq_otg_remove(pdev);
-- 
1.7.5.4

